### ISSUE TITLE
`Diameter.__bytes__` has side effects (messes up `len` and `cmd`

### ISSUE THREAD

**Issue by pfirsich** [kbandla/dpkt#647]
**Describe the bug**
After parsing a Diameter message, calling `__bytes__` (e.g. by printing it)

**To Reproduce**
```
diam = dpkt.diameter.Diameter(data)
print(diam.len) # prints a number
print(diam)
print(diam.len) # prints a bytes object
```

**Expected behavior**
No fields should be affected by printing the object. `len` and `cmd` should stay the same.

**Details(please complete the following information):**
 - OS: Red Hat Enterprise Linux Server 7.9 (Maipo)
 - Python Version: 3.6.6

Sadly I cannot upload any pcap files (legal reasons), but the issue is very obvious.

**Additional context**
The cause of the problem is that `__bytes__` calls `pack_hdr` which changes the member variables. Ideally it would not touch them or at the very least restore them: https://github.com/kbandla/dpkt/blob/master/dpkt/diameter.py#L106

**Comment by obormot** [comment]
Good catch. Assigning `len` and `cmd` in both the unpacking and packing functions definitely seems wrong.
